dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesTracker.kt

```diff
@@ -0,0 +1,55 @@
+package com.datadog.android.rum.gestures
+
+import android.app.Activity
+import androidx.core.view.GestureDetectorCompat
+import com.datadog.android.tracing.Tracer
+import java.lang.ref.WeakReference
+
+internal class DatadogGesturesTracker(val rumTracer: Tracer) :
+    GesturesTracker {
+
+    // region GesturesTracker
+    override fun startTracking(activity: Activity) {
```

Comment:
shouldnt we have a `startTracking(view: View)` rather than `activity` (we could always create one for activity and delegate to the decorView); this way consumer are able to decide what they want to track (and not everything in an app with one activity with plenty of custom views; fragments) ?

Reply:
We don't have access to any view from the LifecycleMonitor. Have in mind that this detector will be added from the LifecycleMonitor, the intention is to have 0 code on the developer side for this logic therefore the only place where we can *hook to*  and *remove from* is in the LifecycleMonitor.

Reply:
(but that would mean handling the painful case were tracking is setup on a view and one/many of its children... )

Reply:
> We don't have access to any view from the LifecycleMonitor. Have in mind that this detector will be added from the LifecycleMonitor, the intention is to have 0 code on the developer side for this logic therefore the only place where we can _hook to_ and _remove from_ is in the LifecycleMonitor.

Not related to this PR and you probably already thought of it but cant we do something with the viewtreeobserver?

Reply:
For now we stick to activity level. The ViewTreeObserver is something that can be used mainly for detecting new Views added (screen views detection), this PR is about gestures detection. Also I've been playing a lot with the ViewTreeObserver in the past, it is something very heavy as every time there's an UI interaction (visibility change, scrolling list, etc.) the ViewTree will change and we don't want add that computation on the MainThread. I would stay away from the ViewTreeObserver for any feature we have for now.